crates/bevy_asset/src/event.rs

```diff
@@ -1,8 +1,49 @@
-use crate::{Asset, AssetId};
+use crate::{Asset, AssetId, AssetLoadError, AssetPath, Handle, UntypedAssetId, UntypedHandle};
 use bevy_ecs::event::Event;
 use std::fmt::Debug;
 
-/// Events that occur for a specific [`Asset`], such as "value changed" events and "dependency" events.
+/// An event emitted when a specific [`Asset`] fails to load.
+#[derive(Event, Clone, Debug)]
+pub struct AssetLoadFailedEvent<A: Asset> {
+    pub id: AssetId<A>,
+    /// The original handle returned when the asset load was requested.
```

Comment:
When will this be None?

Reply:
To be totally honest, I can't say I have a firm grasp on that. It maps 1:1 with `InternalAssetEvent::Failed` – which in some cases doesn't have a resolved handle.

I'd be okay with dropping this field for the time being. Users can use the asset path to lookup the asset id / handle from the asset server if needed and there's no real reason to keep the handle alive via the event. The handle should be in the world already from the original `load()` if it's useful to retain the asset (and potentially retry).

What do you think?